src/slurm_plugin/clustermgtd.py

```diff
@@ -950,6 +963,21 @@ def _find_active_nodes(partitions_name_map):
                 active_nodes += partition.slurm_nodes
         return active_nodes
 
+    @staticmethod
+    def _get_unhealthy_ice_nodes(unhealthy_dynamic_nodes: list) -> dict:
+        """Get insufficient capacity compute resource and nodes, error code mapping."""
+        ice_compute_resources_and_nodes_map = {}
+        for node in unhealthy_dynamic_nodes:
+            if node.is_ice():
+                queue_compute_resourc_name = node.queue_name + "-" + node.compute_resource_name
+                if queue_compute_resourc_name not in ice_compute_resources_and_nodes_map:
+                    ice_compute_resources_and_nodes_map[
+                        queue_compute_resourc_name
+                    ] = InsufficientCapacityComputeResource([node], node.error_code)
```

Comment:
since you are always taking the error_code from the first node you add I'm wondering if we really need this class or we could just store the list of nodes.

Reply:
also you might use the setdefault as you did in other places to remove the if/else

Reply:
Changed to
```
    def _update_insufficient_capacity_compute_resources(
        self, ice_compute_resources_and_nodes_map: Dict[str, Dict[str, List[SlurmNode]]]
    ):
        """Add compute resource to insufficient_capacity_compute_resources if node is ICE node."""
        for queue_name, compute_resources in ice_compute_resources_and_nodes_map.items():
            for compute_resource, nodes in compute_resources.items():
                if not self._insufficient_capacity_compute_resources.get(queue_name, {}).get(compute_resource):
                    self._insufficient_capacity_compute_resources.setdefault(queue_name, {})[
                        compute_resource
                    ] = ComputeResourceFailureEvent(self._current_time, nodes[0].error_code)
```